### ISSUE TITLE
SDK reporting an error (SDK version 14.6.3.0)

### ISSUE THREAD

**Issue by AlexITMS** [intuit/QuickBooks-V3-DotNET-SDK#224]
This morning the following error occurred, this code and system has been functional for months without issue but today we cannot connect at all. I have debugged it as best I can and it does look like the issue is inside the SDK.

Server Error in '/' Application.
Invalid URI or environment
Description: An unhandled exception occurred during the execution of the current web request. Please review the stack trace for more information about the error and where it originated in the code.

Exception Details: System.Exception: Invalid URI or environment

Source Error:

Line 33:         public static string SQLLITEDB = ConfigurationManager.AppSettings["SQLliteDb"];
Line 34: 
Line 35:         public static OAuth2Client auth2Client = new OAuth2Client(CLIENTID, CLIENTSECRET, REDIRECTURL, ENVIRONMENT);
Line 36: 
Line 37: 

Source File: D:\GitHub\ITMSConnectQuick\Controllers\AppController.cs    Line: 35

Stack Trace:

[Exception: Invalid URI or environment]
   Intuit.Ipp.OAuth2PlatformClient.OAuth2Client.GetDiscoveryDoc() in C:\Users\rkasaraneni\QuickBooks-V3-DotNET-SDK\IPPDotNetDevKitCSV3\Code\Intuit.Ipp.OAuth2PlatformClient\Client\OAuth2Client.cs:254
   Intuit.Ipp.OAuth2PlatformClient.OAuth2Client..ctor(String clientID, String clientSecret, String redirectURI, String environment) in C:\Users\rkasaraneni\QuickBooks-V3-DotNET-SDK\IPPDotNetDevKitCSV3\Code\Intuit.Ipp.OAuth2PlatformClient\Client\OAuth2Client.cs:174
   ITMS.Controllers.AppController..cctor() in D:\GitHub\ITMSConnectQuick\Controllers\AppController.cs:35

[TypeInitializationException: The type initializer for 'ITMS.Controllers.AppController' threw an exception.]
   ITMS.Controllers.AppController..ctor() +0

[TargetInvocationException: Exception has been thrown by the target of an invocation.]
   System.RuntimeTypeHandle.CreateInstance(RuntimeType type, Boolean publicOnly, Boolean noCheck, Boolean& canBeCached, RuntimeMethodHandleInternal& ctor, Boolean& bNeedSecurityCheck) +0
   System.RuntimeType.CreateInstanceSlow(Boolean publicOnly, Boolean skipCheckThis, Boolean fillCache, StackCrawlMark& stackMark) +142
   System.Activator.CreateInstance(Type type, Boolean nonPublic) +107
   System.Activator.CreateInstance(Type type) +13
   System.Web.Mvc.DefaultControllerActivator.Create(RequestContext requestContext, Type controllerType) +78

[InvalidOperationException: An error occurred when trying to create a controller of type 'ITMS.Controllers.AppController'. Make sure that the controller has a parameterless public constructor.]
   System.Web.Mvc.DefaultControllerActivator.Create(RequestContext requestContext, Type controllerType) +245
   System.Web.Mvc.MvcHandler.ProcessRequestInit(HttpContextBase httpContext, IController& controller, IControllerFactory& factory) +265
   System.Web.Mvc.MvcHandler.BeginProcessRequest(HttpContextBase httpContext, AsyncCallback callback, Object state) +77
   System.Web.CallHandlerExecutionStep.System.Web.HttpApplication.IExecutionStep.Execute() +1128
   System.Web.HttpApplication.ExecuteStepImpl(IExecutionStep step) +220
   System.Web.HttpApplication.ExecuteStep(IExecutionStep step, Boolean& completedSynchronously) +134

Version Information: Microsoft .NET Framework Version:4.0.30319; ASP.NET Version:4.8.4494.0

**Comment by bvossler** [comment]
I have had the same problem for the past 2 days. Have you found any resolution?

**Comment by bvossler** [comment]
In case you haven't found the fix or if anyone else is looking here...

The problem appears to be to be related to a TLS update on Intuit's side. I cannot confirm with them but I know they had been advertising a TLS update scheduled for July 1. Apparently they did some updating early because I added code to update TLS into my app and that solved the problem. I just added code to ensure TLS 1.2 compatibility using System.Net.ServicePointManager.SecurityProtocol.

Hope this helps someone!
